### ISSUE TITLE
Port 

### ISSUE THREAD

**Issue by lgrkvst** [AlmostInteractive/node-red-contrib-basic-ical-calendar#1]
First off, thanks for making this slimmer version of node-red-contrib-ical-events. :thumbsup:🌷 

I took it for a test drive and stumbled across a few questions:

1. According to the documentation the node should send on port 1 on event start and on port 2 on event end, however all I'm getting is a payload on port 2 (end) at the _start_ of the event. The source code seemingly confirms this to be a bug:
```
    var sendStatus = function (node, calConfigNode) {
        var calculateStatus = function () {
            var inEvent = !calConfigNode.events.every(function (event) {
                var start = basic_ical_events_1.icalCalendar.countdown(event.eventStart);
                var end = basic_ical_events_1.icalCalendar.countdown(event.eventEnd);
                return !((0, node_common_1.inThePast)(start) && (0, node_common_1.inTheFuture)(end));
            });
            if (inEvent) {
                node.send({ payload: { inEvent: inEvent } });
            }
            else {
                node.send([null, { payload: { inEvent: inEvent } }]); // "if outside event
            }
            node.status({
                fill: 'green',
                shape: 'ring',
                text: "".concat(calConfigNode.events.length, " events, in event ").concat(inEvent, ", ").concat(getNextCheckTimeString(node)),
            });
        };
        calculateStatus();
    };
```
A debugging session reveals that when `inEvent` is evaluated, `start` has the value `{days: -0, hours: -0, minutes: -0, seconds: -0}` which `!((0, node_common_1.inThePast)(start) && (0, node_common_1.inTheFuture)(end))` evaluates to true, which ` !calConfigNode.events` then negates...

3. I was curious as to the nodes' resilience but I'm having issues reading the `__generator` code in `cal-config.js`. I understand as much as that you've removed the cache. I tried reploying with a disconnected wifi which crashed my entire NR instance:
```
1 Oct 00:54:13 - [red] Uncaught Exception:
red.js:501
1 Oct 00:54:13 - [error] Error: getaddrinfo ENOTFOUND calendar.google.com
```
Would it be possible to add a try/catch?

On the topic of caching; a quick win could be storing the next five events in a node context variable...

5. Copy-wise, right now my trigger node says `next check Oct 01, 24:30 [14m 3s]`. However I set the calendar config to refresh every hour. Looking at the source code it seems node._nextCheckTime is actually __node._nextTriggerTime__ (?) and consequently the status should read `next trigger Oct 01...`.
6. Right now the trigger node sends `inEvent: true/false`. I think it would make more sense for the trigger to send the event object. If I were interested in inEvent, I could easily serialize the trigger with the sensor component, however sending the event opens the door to having Node Red act on data inside calender events.

Awesome work so far! I went through all the calendar components in the node red flow library and found this to be easily the 🥇 option if you want something user-friendly that (almost 😃) just works.

**Comment by AlmostInteractive** [comment]
@lgrkvst 
Thanks for the comments!  It looks like you're on an old version of the library though.  Even though you're looking at the generated javascript and not the source typescript (reading the TS will be _much_ easier, I promise!) I can see that the `var inEvent` is being calculated with the `countdown` values which I got rid of.  (see https://github.com/AlmostInteractive/node-red-contrib-basic-ical-calendar/blob/master/src/nodes/node-common.ts#L29)  I'll address the topics that I can confirm are still valid on the latest version anyway.  😃

3. Done.  Good catch.  Now it reports an error to NodeRed instead of being an uncaught error which crashes NodeRed.  Thank you!

5. "check" as in "check the status of", however you have a valid point, it triggers as a result of checking.  That's a bug, in that you'll get an `{inEvent: false}` message if you have nothing scheduled within a day of the last event.  I can fix that.

6. That's an easy upgrade.  I didn't add it because I won't use it, but I can stick that in for you since you took the time to write me.  😄 I'm going to add it to the Sensor, though, not the Trigger.  It just made sense to me.  Your Trigger and activate a Sensor to get the info, though, so you'll be ok!

So yeah, grab  the latest version (v2.1.0 @ https://flows.nodered.org/node/@almostinteractive/node-red-contrib-basic-ical-calendar) and try again!

**Comment by lgrkvst** [comment]
Haha, I wrote that Saturday night and a fix came Sunday morning. Swift like a cobra! 🐍 

The oddest thing – I installed the repo yesterday evening for the first time and for some reason got version 2.0.1...

Anyway great! The trigger node triggers just like advertised.

However (there's always a however), the sensor reports `inEvent: true` no matter when I trigger it. 

The trigger node still says `next check` – I think you can avoid some confusion by changing the nomenclature. Granted the trigger node does check when to fire, but it's the triggering that is of value and the user probably don't want to know about its inner workings.

Great that it's sending event information, however it's sending ALL of the events so I still have to perform just the kind of logic I want to outsource to the calendar component in order to determine what calendar data triggered an event.

Might I suggest an information architecture that minimizes "collateral logic":
* sensor node to send just the current `event` as payload if there's an ongoing event (analogous to literally "sensing" the calendar) and false as payload if not.
* trigger node to just send `event` as payload. `inEvent` doesn't make any sense as the trigger node cannot trigger unless there's an event. Right now the trigger node sends `inEvent: true` via its _end_ port which is a philosophic discussion. By just sending `event` at the start and at the end, a user/flow can determine start/end from the separation of the ports.
* a dedicated node to obtain upcoming events.

~~A last question: what has changed with the formatting? It's not minified or obscured, but for some reason all the line breaks and indents are gone.~~ (edit: just realized you're transpiling from ts)

**Comment by AlmostInteractive** [comment]
@lgrkvst 
1. Not sure why it's still telling you v2.0.1.  Sounds like your system hasn't grabbed the latest repo information.  Maybe try restarting the service?

2. I'm not able to reproduce the Sensor always reporting `inEvent: true`.  If you'd like to sync up on Discord and do a screen share session, I'd be curious to see that.  Send me your id and I'll contact you. 
![image](https://github.com/AlmostInteractive/node-red-contrib-basic-ical-calendar/assets/3893631/4302988c-3d56-4959-8963-0c85ce078e7d)

3. Since the Sensor can be triggered whenever (not just an event beginning / ending threshold) it made sense to me to send all currently running events.  I think that's what you're asking for anyway.  (Rather than making `events` false if there are none, I prefer to leave it as an empty array.  I see no value in sending two false values and I like simple return types.  😄 )

4. For a Trigger node, `inEvent` is useful for me since all I, personally, only care about is whether or not I'm currently in an event.  (I use this to turn on / off my background music.)  Sending _just_ an event wouldn't make too much sense for me since then I'd need to do some additional work (in a JS node probably) to see if the event is starting or ending.  I can probably send along the `event` in the payload, though I'm not certain what will happen if there are two events that have overlapping thresholds (eg: two events start or end at the same time, or one starts when another ends).  I'll have to test that and make sure it makes sense.  I wanted to keep the Trigger node light, pretty much a little guy yelling "Hey, something happened!"  To do anything beyond that, use the Sensor node.

5.  I like the idea of a node to obtain the next N events.  Feel like writing a new feature?  😁

**Comment by AlmostInteractive** [comment]
@lgrkvst 
2. I started looking around and yup, I made a silly mistake.  In some situations it'll report `inEvent: true` when indeed it is false.  I'll fix that bug in the next update.

Released v2.2.0 that, principally, emits an event start and end output along with the triggering event for every event.  Hope it helps.  😄
